sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -5,114 +5,220 @@
 
 import com.azure.core.util.logging.ClientLogger;
 import org.reactivestreams.Subscription;
+import reactor.core.Disposable;
 import reactor.core.publisher.BaseSubscriber;
+import reactor.core.publisher.Mono;
 
-import java.util.Objects;
-import java.util.Timer;
-import java.util.TimerTask;
+import java.time.Duration;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
 
 /**
  * Subscriber that listens to events and publishes them downstream and publishes events to them in the order received.
  */
 class SynchronousMessageSubscriber extends BaseSubscriber<ServiceBusReceivedMessageContext> {
     private final ClientLogger logger = new ClientLogger(SynchronousMessageSubscriber.class);
-    private final Timer timer = new Timer();
     private final AtomicBoolean isDisposed = new AtomicBoolean();
-    private final SynchronousReceiveWork work;
+    private final long prefetch;
+    private final AtomicInteger wip = new AtomicInteger();
+    private final Queue<SynchronousReceiveWork> workQueue = new ConcurrentLinkedQueue<>();
+    private final Queue<ServiceBusReceivedMessageContext> bufferMessages = new ConcurrentLinkedQueue<>();
+    private final SynchronousReceiveWork initialWork;
+    private final AtomicLong remaining = new AtomicLong();
 
     private volatile Subscription subscription;
 
-    SynchronousMessageSubscriber(SynchronousReceiveWork work) {
-        this.work = Objects.requireNonNull(work, "'work' cannot be null.");
+    private SynchronousReceiveWork currentWork;
+    private Disposable timeoutOperation;
+
+    SynchronousMessageSubscriber(long prefetch, SynchronousReceiveWork initialWork) {
```

Comment:
Why not queue the initial request? You won't need the `initialWork` as a member variable. And you can use the prefetch.

```java
    SynchronousMessageSubscriber(long prefetch, SynchronousReceiveWork initialWork) {
        this.prefetch = prefetch;
        this.workQueue.add(initialWork)
    }

    @Override
    protected void hookOnSubscribe(Subscription subscription) {
        this.subscription = subscription;
        subscription.request(prefetch);
    }
```